HaloDAO/halodao-interface
# Bridge Integration

## Description of Changes


[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDEV-183)
[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDEV-181)

1. Token Approvals Integration
2. Deposit Bridge Integration
3. Burn Wrapped Tokens Integration
4. Switching Bridge Origin/Destination feature
5. Choosing Token to Bridge feature
6. Button state changes react to current balance/allowance

### How to test:
1. Update you env file from [1Password](https://start.1password.com/open/i?a=2VIA62LB7VAW3NACLOHTFW4ERI&h=halodao.1password.com&i=6bzbszvhwrbm7lztrvp5sfhfy4&v=gllrg2so74fsgd43baj5q5voui) - (If you're testing locally)
2. Secure MKT (Mock Token) from Brent. 
3. Secure Matic tokens for your gas -> https://matic.supply/
3. Go to /Bridge.
4. Make sure you are in Polygon Mainnet.
5. Default network will be selected (in this case Polygon -> BSC).
6. Enter amount of MKT you want to bridge.
7. Approve button should appear.
8. Once approve tx is confirmed, "Next" button will appear, click "Next" to bridge the token.
9. Confirm the tx in Metamask.
10. Confirm transaction modal should appear.
11. When transaction is confirmed, your MKT balance is polygon should be updated.
12. To check your WMKT balance, select BSC as the source network.
13. WMKT should be selected from the currency input.
14. Your WMKT balance should be updated only after the transaction is considered final. (After 192 blocks for polygon - you can check polygonscan)

### Developer Checklist:


### Reviewers Checklist:
